ext/standard/php_crypt_r.c

```diff
@@ -154,9 +154,7 @@ char * php_md5_crypt_r(const char *pw, const char *salt, char *out) {
 	}
 
 	/* It stops at the first '$', max 8 chars */
-	for (ep = sp; *ep != '\0' && *ep != '$' && ep < (sp + 8); ep++) {
-		continue;
-	}
+	for (ep = sp; *ep != '\0' && *ep != '$' && ep < (sp + 8); ep++) {}
```

Comment:
Does C support the `expression` + `semicolon` syntax?
```suggestion
	for (ep = sp; *ep != '\0' && *ep != '$' && ep < (sp + 8); ep++);
```

Reply:
Yes, C supports such syntax also. I'm not sure what would be a good approach here. In general, reading the code might be most efficient where the loop body is filled with something that changes variable value. Not that this is followed here or anything, but it might be a bit of a hint what coding styles use:
https://google.github.io/styleguide/cppguide.html#Loops_and_Switch_Statements
(section under empty loop body has a minor explanation) about performance and all, it's pretty much all the same probably.

Also, the clang-format tool doesn't have any such option so far that might sync this across the code (theoretically speaking) so it's pretty much an internal code style pick where to go here.

Reply:
So the proposed change `for (...) {}` is based on a rule in `clang-tidy`: https://clang.llvm.org/extra/clang-tidy/checks/readability-redundant-control-flow.html, if that helps.

Reply:
`for (...);` is the syntax we'd usually use here. It's not terribly common but you'll find a handful of such loops strewn across the codebase.

Reply:
Thanks for the note @nikic. I did a pass over the whole code base and fixed these cases to use the `for (...);` syntax.